### ISSUE TITLE
custom-event-name-casing flagging valid naming as invalid

### ISSUE THREAD

**Issue by derick-montague** [vuejs/eslint-plugin-vue#1340]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7.11.0
- **eslint-plugin-vue version:** 7.1.0
- **Node version:** 12.1.x
- **Operating System:** MacOS

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: ['plugin:vue/recommended', 'eslint:recommended', '@vue/prettier'],
  rules: {
    'no-console': 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'prettier/prettier': [
      'error',
      {
        singleQuote: true,
        trailingComma: 'none'
      }
    ],
    'vue/component-name-in-template-casing': ['error', 'kebab-case']
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  globals: {
    expect: true,
    sinon: true
  },
  overrides: [
    {
      files: [
        '**/__tests__/*.{j,t}s?(x)',
        '**/tests/unit/**/*.spec.{j,t}s?(x)'
      ],
      env: {
        jest: true
      }
    }
  ]
};

```

**What did you do?**	

```vue
    toggleNavigation() {
      this.$root.$emit('toggle:navigation');
    }
```

**What did you expect to happen?**
The custom event naming should be valid

**What actually happened?**
`Custom event name 'toggle:navigation' must be kebab-case.eslintvue/custom-event-name-casing`

The issue seems to be   [`return isKebabCase(name) || name.startsWith('update:')`](https://github.com/vuejs/eslint-plugin-vue/blob/80b8983118cc08cf473642c05413bcb930c93613/lib/rules/custom-event-name-casing.js#L26)

The short-circuit is expecting the naming to start with `update:`, rather than a colon preceded by any lower case word.

**Repository to reproduce this issue**
N/A

**Comment by derick-montague** [comment]
I see this is expected behavior.

**Comment by derick-montague** [comment]
Thank you for all the effort!
